chain/client/src/view_client.rs

```diff
@@ -246,6 +250,62 @@ impl ViewClientActor {
         }
     }
 
+    /// Returns maintenance windows by account.
+    fn get_maintenance_windows(
+        &self,
+        account_id: AccountId,
+        genesis_config: GenesisConfig,
+    ) -> Result<MaintenanceWindowsView, near_chain::Error> {
+        let epoch_manager = EpochManager::new_from_genesis_config(
+            self.runtime_adapter.store().clone(),
+            &genesis_config,
+        )?;
+        let chain_store = ChainStore::new(
+            self.runtime_adapter.store().clone(),
+            genesis_config.genesis_height,
+            false,
+        );
+
+        let head = chain_store.head()?;
+        let epoch_id = epoch_manager.get_epoch_id(&head.last_block_hash)?;
+        let epoch_info: Arc<EpochInfo> = epoch_manager.get_epoch_info(&epoch_id)?;
+        let num_shards = self.runtime_adapter.num_shards(&epoch_id)?;
+        let cur_block_info = epoch_manager.get_block_info(&head.last_block_hash)?;
+        let next_epoch_start_height = epoch_manager
+            .get_epoch_start_height(cur_block_info.hash())?
+            + epoch_manager.get_epoch_config(&epoch_id)?.epoch_length;
+
+        let mut windows: MaintenanceWindowsView = Vec::new();
+        let mut start_block_of_window: Option<BlockHeight> = None;
+        let last_block_of_epoch = next_epoch_start_height - 1;
+
+        for block_height in head.height..next_epoch_start_height {
+            let bp = epoch_info.sample_block_producer(block_height);
+            let bp = epoch_info.get_validator(bp).account_id().clone();
+            let cps: Vec<AccountId> = (0..num_shards)
+                .into_iter()
+                .map(|shard_id| {
+                    let cp = epoch_info.sample_chunk_producer(block_height, shard_id);
+                    let cp = epoch_info.get_validator(cp).account_id().clone();
+                    cp
+                })
+                .collect();
+            if account_id != bp && !cps.iter().any(|a| *a == account_id) {
+                if let Some(start) = start_block_of_window {
+                    if block_height == last_block_of_epoch {
+                        windows.push((start, block_height + 1));
+                    }
+                } else {
+                    start_block_of_window = Some(block_height);
+                }
+            } else if let Some(start) = start_block_of_window {
+                windows.push((start, block_height + 1));
```

Comment:
this seems incorrect.

If you're a producer for block_height -- the maintenance window should finish at 'block_height', right?

Reply:
Yes, this also confuses me.  

At first, I also implement `(1,2)` for the first block and the second block.  And some people think it may be misleading for others because the Rust `Range` object is `Range(1, 3)` for the same scenario. So I change the implementation as similar to `Range` object. Also, add the example as following

https://github.com/near/docs/pull/1277/files#diff-6ed641db4bb40f6ce8ab6fd360504141e710d6529f538d04adf7b1b4eddfd6f8R54

Should I change this back? If it is much more common in JSON format.

Reply:
You can also use Range type directly.  Change MaintenanceWindowsView from `Vec<(BlockHeight, BlockHeight)>` to `Vec<Range<BlockHeight>>` and semantics of it will be clear.  You might need to implement custom serde serialiser for it though.

Reply:
Sure, we can do that. Thanks.